apps/passport-server/src/services/userService.ts

```diff
@@ -260,6 +269,66 @@ export class UserService {
     res.sendStatus(200);
   }
 
+  /**
+   * Checks whether allowing a user to reset their account one more time
+   * would cause them to exceed the account reset rate limit. If it does,
+   * throws an error. If it doesn't, saves this account reset timestamp
+   * to the database and proceeds. Can only be called on users that have
+   * already created an account.
+   */
+  private async checkAndIncrementAccountRateLimit(
+    user: CommitmentRow
+  ): Promise<void> {
+    if (process.env.ACCOUNT_RESET_RATE_LIMIT_DISABLED === "true") {
+      logger("[PCDPASS] account rate limit disabled");
+      return;
+    }
+
+    const now = Date.now();
+    const configuredRateLimitDurationMs = parseInt(
+      process.env.ACCOUNT_RESET_LIMIT_DURATION_MS ?? "",
+      10
+    );
+    const configuredAccountResetQuantity = parseInt(
+      process.env.ACCOUNT_RESET_LIMIT_QUANTITY ?? "",
+      10
+    );
+    const defaultRateLimitDurationMs = 1000 * 60 * 60 * 24; // default 24 hours
+    const defaultRateLimitQuantity = 5; // default max 5 resets (not including 1st time account creation) in 24 hours
+    const rateLimitDurationMs = isNaN(configuredRateLimitDurationMs)
+      ? defaultRateLimitDurationMs
+      : configuredRateLimitDurationMs;
+    const rateLimitQuantity = isNaN(configuredAccountResetQuantity)
+      ? defaultRateLimitQuantity
+      : configuredAccountResetQuantity;
+
+    const parsedTimestamps: number[] = user.account_reset_timestamps.map((t) =>
```

Comment:
How/when does this user object and its associated account_reset_timestamps column get updated after updateCommitmentResetList below?

Reply:
good catch, at the time you reviewed this, there was a bug. now it updates in the case we're not over the rate limit, as indicated by the `exceedsRateLimit` variable